neuralprophet/TorchProphet.py

```diff
@@ -0,0 +1,553 @@
+        model. If mcmc.samples>0, this will be integrated over all model
+        parameters, which will include uncertainty in seasonality.
+    uncertainty_samples: Not required for NeuralProphet.
+    stan_backend: Not supported by NeuralProphet.
+    """
+
+    def __init__(
+        self,
+        growth="linear",
+        changepoints=None,
+        n_changepoints=25,
+        changepoint_range=0.8,
+        yearly_seasonality="auto",
+        weekly_seasonality="auto",
+        daily_seasonality="auto",
+        holidays=None,
+        seasonality_mode="additive",
+        seasonality_prior_scale=None,
+        holidays_prior_scale=None,
+        changepoint_prior_scale=None,
+        mcmc_samples=None,
+        interval_width=0.80,
+        uncertainty_samples=None,
+        stan_backend=None,
+        **kwargs,
+    ):
+        # Check for unsupported features
+        if seasonality_prior_scale or holidays_prior_scale or changepoint_prior_scale:
+            log.info(
+                "Using `_prior_scale` is unsupported for regularisation in NeuralProphet, please use the `_reg` args instead."
+            )
+        if mcmc_samples or uncertainty_samples:
+            log.info(
+                "Providing the number of samples for Bayesian inference or Uncertainty estimation is not required in NeuralProphet."
+            )
+        if stan_backend:
+            log.info("A stan_backend is not used in NeuralProphet. Please remove the parameter")
+        # Run the NeuralProphet function
+        super(TorchProphet, self).__init__(
+            growth=growth,
+            changepoints=changepoints,
+            n_changepoints=n_changepoints,
+            changepoints_range=changepoint_range,
+            yearly_seasonality=yearly_seasonality,
+            weekly_seasonality=weekly_seasonality,
+            daily_seasonality=daily_seasonality,
+            seasonality_mode=seasonality_mode,
+            # TODO: activate once PR merged prediction_interval=interval_width,
+            **kwargs,
+        )
+        # Handle holidays as events
+        if holidays is not None:
+            self.add_events(
+                events=list(holidays["holiday"].unique()),
+                lower_window=holidays["lower_window"].max(),
+                upper_window=holidays["upper_window"].max(),
+            )
+            self.events_df = holidays.copy()
+            self.events_df.rename(columns={"holiday": "event"}, inplace=True)
+            self.events_df.drop(["lower_window", "upper_window"], axis=1, errors="ignore", inplace=True)
+
+        # Overwrite NeuralProphet properties
+        self.name = "Prophet"
+        self.history = None
+
+        # Unused properties
+        self.train_holiday_names = None
+
+    def validate_inputs(self):
+        """
+        Validates the inputs to Prophet.
+        """
+        raise NotImplementedError("Not required in NeuralProphet as all inputs are automatically checked.")
```

Comment:
instead of an error, we could just log an error and move on, as this should not cause any issues?